addons/mail/models/mail_link_preview.py

```diff
@@ -33,12 +35,17 @@ def _create_from_message_and_notify(self, message):
             return self
         urls = OrderedSet(html.fromstring(message.body).xpath('//a[not(@data-oe-model)]/@href'))
         link_previews = self.env['mail.link.preview']
+        ignore_list = self.env["ir.config_parameter"].sudo().get_param("mail.mail_link_preview_ignore_list").split("\n")
+        ignore_list = "|".join(ignore_list)
```

Comment:
I'm not sure to see what's the point of splitting `\n` and adding `|`.
Why not directly have `|` in the config then?

Reply:
It's to be able to build a simple single-line list inside the config_parameter and merge all the regexp with `or`:
```
.*odoo.com/(web|odoo)
.*facebook.com.*
```
It makes the regexp list manageable for the admin if it starts to grow.

Reply:
The way it is coded, isn't the '|' replace potentially conflicting with the priority of operators on each specific line?
I don't think joining is a good idea, it would probably be better to do separate matches.

Reply:
I guess you are right, I changed to individually test every regexp